torch/fx/experimental/param_fetch.py

```diff
@@ -0,0 +1,42 @@
+from torch.fx.graph_module import GraphModule
+from typing import Any, Dict, List, Tuple
+import torch.nn as nn
+
+
+"""This dict maps the nn.Module class name to the attribute name list that we want to fetch for lowering.
+The first integer in the tuple is the version number of the nn.Module class when we create the parameter list.
+If there's a version mismatch then it means the parameter names in the book might be mismatched with nn.Module.
+"""
+module_fetch_book: Dict[str, Tuple[int, List[str]]] = {
+    "Linear": (1, ["weight", "bias"]),
+    "Conv2d": (1, ["weight", "bias", "kernel_size", "stride", "padding", "dilation", "groups", "padding_mode"]),
+    "BatchNorm2d": (2, ["weight", "bias", "running_mean", "running_var"]),
+}
+
+def fetch_module(mod: nn.Module) -> Dict[str, Any]:
+    """If `mod` is in `module_fetch_book`, fetch the mod's attributes that in the `module_fetch_book`
+    after checking module's version is compatible with the `module_fetch_book`.
+    """
+    parameters: Dict[str, Any] = {}
+    parameters["name"] = type(mod).__name__
+
+    if type(mod).__name__ in module_fetch_book:
+        _version, param_to_fetch = module_fetch_book[type(mod).__name__]
+        if _version != mod._version:
+            print(f"Fetcher version {_version} try to fetch {type(mod).__name__} version {mod._version}")
```

Comment:
Is this an error condition? If so,

```
            raise RuntimeError(f"Fetcher version {_version} try to fetch {type(mod).__name__} version {mod._version}")
```

If not, use `warnings.warn`.

Reply:
We should also be checking that the `Module` we're processing has `version` at *or below* the one we support. Right now, with the 3 modules supported (Linear, Conv2d, and BatchNorm2d) it doesn't matter because the attributes of interest are all the same, but in the future we may need `module_fetch_book` to also contain functions that normalize the retrieved attributes across older versions

Reply:
>contain functions that normalize the retrieved attributes across older versions

Sorry I'm a little bit confused. Do you mean when we processing `Module` with older versions, the function would come out and match the names of attribute in current version to the older version?

Reply:
Yes, that's correct

Reply:
I've added a function to each entry. Right now for these three modules the function doesn't actually do anything.